src/client/Microsoft.Identity.Client/Http/HttpManager.cs

```diff
@@ -54,8 +73,20 @@ public async Task<HttpResponse> SendRequestAsync(
             X509Certificate2 bindingCertificate,
             Func<HttpRequestMessage, X509Certificate2, X509Chain, SslPolicyErrors, bool> validateServerCert,
             CancellationToken cancellationToken,
+            IRetryPolicy retryPolicy = null,
             int retryCount = 0)
         {
+            // Use the default STS retry policy if the request is not for managed identity
+            // and a non-default STS retry policy is not provided.
+            // Skip this if statement the dev indicated that they do not want retry logic.
+            if (!_isManagedIdentity && retryPolicy == null && _withRetry)
+            {
+                retryPolicy = new LinearRetryPolicy(
```

Comment:
nit: Consider creating a RetryPolicyFactory that is responsible for creating all these objects. It can also encapsulate the `HttpRetryConditions` logic.

Reply:
I'll do that in another PR. This one is already big enough.

Reply:
I'll make a separate PR for a RetryPolicyFactory since this PR is so big already.